### PR TITLE
Add surround check to endpoint

### PR DESCRIPTION
**Pull request by shayzluf**
[Part of] #3133

---

# Description
Add surround detection to the IsSlashableAttestation endpoint

- [x] add surround/surrounded detection
- [x] add surround/surrounded retrieval 
- [x] add tests

### CODE REVIEW

File: slasher/rpc/server.go [hunk 1]
```diff
@@ -87,3 +98,40 @@ func (ss *Server) SlashableAttestations(req *types.Empty, server ethpb.Slasher_S
 	//TODO(3133): implement stream provider for newly discovered listening to slashable attestation.
 	return status.Error(codes.Unimplemented, "not implemented")
 }
+
+// DetectSurround is a method used to return the attestation that were detected
+// by min max surround detection method.
+func (ss Server) DetectSurround(ctx context.Context, source uint64, target uint64, validatorIdx uint64) ([]*ethpb.IndexedAttestation, error) {
```
Comment (terencechain) [inline slasher/rpc/server.go]:
```suggestion
func (ss Server) DetectSurroundVotes(ctx context.Context, source uint64, target uint64, validatorIdx uint64) ([]*ethpb.IndexedAttestation, error) {
```

File: slasher/rpc/server.go [hunk 2]
```diff
@@ -87,3 +98,40 @@ func (ss *Server) SlashableAttestations(req *types.Empty, server ethpb.Slasher_S
...
+	if err != nil {
+		return nil, err
+	}
+	maxTargetEpoch, err := ss.DetectAndUpdateMaxEpochSpan(ctx, source, target, validatorIdx)
+	if err != nil {
+		return nil, err
+	}
+	var iAtts []*ethpb.IndexedAttestation
+	if minTargetEpoch > 0 {
+		iats, err := ss.SlasherDB.IndexedAttestation(minTargetEpoch, validatorIdx)
```
Comment (terencechain) [inline slasher/rpc/server.go]: not a fan of `iats`, something better?

File: slasher/rpc/server.go (hunk 1 above)
Comment (shayzluf) [reply slasher/rpc/server.go]: thanks, changed

File: slasher/rpc/server.go (hunk 2 above)
Comment (nisdas) [reply slasher/rpc/server.go]: +1

File: slasher/rpc/server.go [hunk 3]
```diff
@@ -87,3 +98,40 @@ func (ss *Server) SlashableAttestations(req *types.Empty, server ethpb.Slasher_S
...
+			return nil, err
+		}
+		for _, ia := range iats {
+			if ia.Data.Source.Epoch > source && ia.Data.Target.Epoch < target {
+				iAtts = append(iAtts, ia)
+			}
+		}
+	}
+	if maxTargetEpoch > 0 {
+		iats, err := ss.SlasherDB.IndexedAttestation(maxTargetEpoch, validatorIdx)
```
Comment (nisdas) [inline slasher/rpc/server.go]: same here

File: slasher/rpc/server.go (hunk 2 above)
Comment (shayzluf) [reply slasher/rpc/server.go]: changed

File: slasher/rpc/server.go (hunk 3 above)
Comment (shayzluf) [reply slasher/rpc/server.go]: thanks. changed

**Pull request merged**